test/screenshot/screenshot.js

```
@@ -1,60 +1,220 @@
+import {Readable} from 'stream';
+import {createHash} from 'crypto';
+import {readFile, writeFile} from 'fs';
+import {promisify} from 'util';
 import puppeteer from 'puppeteer';
 import compareImages from 'resemblejs/compareImages';
-import {promisify} from 'util';
-import {readFile, writeFile} from 'fs';
 import {assert} from 'chai';
+import Storage from '@google-cloud/storage';
 
 import comparisonOptions from './screenshot-comparison-options';
 
 const readFilePromise = promisify(readFile);
 const writeFilePromise = promisify(writeFile);
 
+const SERVICE_ACCOUNT_KEY = process.env.SERVICE_ACCOUNT_KEY;
+const BRANCH_NAME = process.env.BRANCH_NAME;
+const COMMIT_HASH = process.env.COMMIT_HASH;
+const GOLDENS_FILE_PATH = './test/screenshot/golden.json';
+const BUCKET_NAME = 'screenshot-uploads';
+const DEFAULT_METADATA = {
+  commit: COMMIT_HASH,
+  branch: BRANCH_NAME,
+};
+
+const GOLDEN = 'golden';
+const SNAPSHOT = 'snapshot';
+const DIFF = 'diff';
+
+let credentials;
+try {
+  credentials = JSON.parse(SERVICE_ACCOUNT_KEY);
+} catch (err) {
+  console.error('Service account key could not be parsed.');
+  process.exit(1);
```

This will break any future unit tests. Maybe just rethrow the error instead?